File: src/test/java/io/tripled/social/client/application/SpiedPostMessageUseCase.java
package io.tripled.social.client.application;

public class SpiedPostMessageUseCase implements PostMessageUseCase {

  public String userName;
  public String message;

  @Override
  public String postMessage(String userName, String message) {
    this.userName = userName;
    this.message = message;

    return userName + " posted: " + message;
  }
}

File: src/test/kotlin/io/tripled/social/client/application/SpiedPostMessageUseCase.kt
package io.tripled.social.client.application

class SpiedPostMessageUseCase : PostMessageUseCase {
    @JvmField
    var userName: String? = null
    @JvmField
    var message: String? = null

    override fun postMessage(userName: String, message: String): String {
        this.userName = userName
        this.message = message

        return userName + " posted: " + message
    }
}
